src/onegov/town6/api.py

```diff
@@ -215,31 +217,51 @@ def for_filter(self, **filters: Any) -> Self:
 
         """
 
-        return self.__class__(self.request, self.directory, filters)
+        return self.__class__(self.request, self.endpoint, filters)
 
     def by_id(
             self, id: PKType
               ) -> ExtendedDirectoryEntry | None:
         """ Return the item with the given ID from the collection. """
 
         try:
-            return self.__class__(self.request, self.directory
+            return self.__class__(self.request, self.endpoint
                                   ).collection.by_id(id)
         except SQLAlchemyError:
             return None
 
     def item_data(self, item: ExtendedDirectoryEntry) -> dict[str, Any]:
-        data = item.content['values'].copy() or {}
-        data['coordinates'] = get_geo_location(item)
-
-        for field in item.directory.fields:
-            if any(type in field.type for type in ['fileinput', 'url']):
-                data.pop(field.id, None)
+        data: dict[str, Any] = {}
+
+        if item.access == 'public':
+            if item.content_fields:
+                data = {f.name: f.object_data for f in item.content_fields}
+
+            for field in item.directory.fields:
+                if any(field_type in field.type for field_type in [
+                    'fileinput', 'url']):
+                    data.pop(field.id, None)
+                if any(field_type in field.type for field_type in [
+                    'date', 'datetime']):
+                    if data.get(field.id):
+                        data[field.id] = data[field.id].isoformat()
+
+            data['coordinates'] = get_geo_location(item)
+        elif item.access == 'mtan':
+            data['title'] = item.title
+            data['lead'] = item.lead
 
         return data
 
     def item_links(self, item: ExtendedDirectoryEntry) -> dict[str, Any]:
-        data = {(f.note or 'file'): f for f in item.files.copy()}
+        data = {}
+        if item.access == 'public':
+            content_field_names = []
+            if item.content_fields:
+                content_field_names = [i.name for i in item.content_fields]
+            data = {(f.note or 'file'): f for f in item.files
+            if f.note and f.note.split(':')[0] in content_field_names}
```

Comment:
```suggestion
            data = {
                (file.note or 'file'): file
                for file in item.files
                if file.note
                if file.note.split(':', 1)[0] in content_field_names
            }
```
The formatting made this a little hard to read, i like putting each part of the comprehension on a separate line when the conditions get complex enough. I also don't recall what kind of upload field we're using in formcode, there's a possibility the file is accessible from the field itself, which would simplify the code a bit.

I think you also forgot to include the url field values in these items.